egs/iam/v1/local/extract_feature.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+# Copyright   2017 Yiwen Shao
+
+nj=4
+cmd=run.pl
+feat_dim=40
+echo "$0 $@"
+
+. ./cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh || exit 1;
+
+data=$1
+featdir=$data/data
+scp=$data/images.scp
+logdir=$data/log
+
+mkdir -p $logdir
+mkdir -p $featdir
+
+# make $featdir an absolute pathname
+featdir=`perl -e '($dir,$pwd)= @ARGV; if($dir!~m:^/:) { $dir = "$pwd/$dir"; } print $dir; ' $featdir ${PWD}`
+
+for n in $(seq $nj); do
+    split_scps="$split_scps $logdir/images.$n.scp"
+done
+
+# split images.scp
+echo $split_scps
+utils/split_scp.pl $scp $split_scps || exit 1;
+
+# create a job array of make_features.py
+$cmd JOB=1:$nj $logdir/extract_feature.JOB.log \
+  local/make_features.py $logdir --feat-dim $feat_dim --job JOB \| \
```

Comment:
This is not the best way to do this. Please change make_features.py so that it takes `images.scp` as an argument. Here you can pass `$logdir/images.JOB.scp` to it. You can also remove the `dir` argument and add a new optional argument `--allowed-lengths-file` (which is an empty string by default) and set it if you find the `allowed_lengths.txt` file in `$data`, e.g. if [ -f $data/allowed_lengths.txt] ...